### PR TITLE
Feature/481280 handle run exception

### PR DESCRIPTION
**Pull request by udaydefra**

### CODE REVIEW

**Review by johnbenjamin-defra — DISMISSED** [review]

File: src/EPR.Calculator.API.UnitTests/CalcResultTests.cs [hunk 1]
```diff
@@ -86,13 +87,13 @@ public void Setup()
         }
 
         [TestMethod]
-        public void PrepareCalcResults_ShouldReturnCreatedStatus()
+        public async void PrepareCalcResults_ShouldReturnCreatedStatus()
```
Comment (tim-powell-eviden) [inline src/EPR.Calculator.API.UnitTests/CalcResultTests.cs:90]: Test should return a Task

File: src/EPR.Calculator.API.UnitTests/Controllers/CalculatorInternalControllerTests.cs [hunk 2]
```diff
@@ -202,7 +203,7 @@ public void UpdateRpdStatus_With_RunId_When_Successful()
         }
 
         [TestMethod]
-        public void PrepareCalcResults_ShouldReturnCreatedStatus()
+        public async void PrepareCalcResults_ShouldReturnCreatedStatus()
```
Comment (tim-powell-eviden) [inline src/EPR.Calculator.API.UnitTests/Controllers/CalculatorInternalControllerTests.cs:206]: Test should return a Task

File: src/EPR.Calculator.API.UnitTests/Controllers/CalculatorInternalControllerTests.cs [hunk 3]
```diff
@@ -214,19 +215,20 @@ public void PrepareCalcResults_ShouldReturnCreatedStatus()
...
+            var result = await controller.PrepareCalcResults(requestDto) as ObjectResult;
             var calculatorRun = dbContext.CalculatorRuns.SingleOrDefault(run => run.Id == 1);
             Assert.IsNotNull(result);
             Assert.AreEqual((int)RunClassification.UNCLASSIFIED, calculatorRun?.CalculatorRunClassificationId);
             Assert.AreEqual(201, result.StatusCode);
         }
 
         [TestMethod]
-        public void PrepareCalcResults_ShouldReturnNotFound()
+        public async void PrepareCalcResults_ShouldReturnNotFound()
```
Comment (tim-powell-eviden) [inline src/EPR.Calculator.API.UnitTests/Controllers/CalculatorInternalControllerTests.cs:231]: Test should return a Task

File: src/EPR.Calculator.API.UnitTests/PrepareCalcResultsValidationTest.cs [hunk 4]
```diff
@@ -34,14 +34,16 @@ public void SetUp()
...
             _transposePomAndOrgDataService = new Mock<ITransposePomAndOrgDataService>();
-            controller = new CalculatorInternalController(_context, _rpdStatusDataValidator.Object, _wrapper.Object, _builder.Object, _exporter.Object, _transposePomAndOrgDataService.Object);
+            var mockStorageService = new Mock<IStorageService>();
+            controller = new CalculatorInternalController(_context, _rpdStatusDataValidator.Object, _wrapper.Object,
+                _builder.Object, _exporter.Object, _transposePomAndOrgDataService.Object, mockStorageService.Object);
         }
 
         [TestMethod]
-        public void PrepareCalcResults_Invalid_RunId()
+        public async void PrepareCalcResults_Invalid_RunId()
```
Comment (tim-powell-eviden) [inline src/EPR.Calculator.API.UnitTests/PrepareCalcResultsValidationTest.cs:43]: Test should return a Task

File: src/EPR.Calculator.API.UnitTests/CalcResultTests.cs (hunk 1 above)
Comment (udaydefra) [reply src/EPR.Calculator.API.UnitTests/CalcResultTests.cs]: Resolved

File: src/EPR.Calculator.API.UnitTests/Controllers/CalculatorInternalControllerTests.cs (hunk 2 above)
Comment (udaydefra) [reply src/EPR.Calculator.API.UnitTests/Controllers/CalculatorInternalControllerTests.cs]: Resolved

File: src/EPR.Calculator.API.UnitTests/Controllers/CalculatorInternalControllerTests.cs (hunk 3 above)
Comment (udaydefra) [reply src/EPR.Calculator.API.UnitTests/Controllers/CalculatorInternalControllerTests.cs]: Resolved

File: src/EPR.Calculator.API.UnitTests/PrepareCalcResultsValidationTest.cs (hunk 4 above)
Comment (udaydefra) [reply src/EPR.Calculator.API.UnitTests/PrepareCalcResultsValidationTest.cs]: Resolved

**Review by johnbenjamin-defra — APPROVED** [review]

**Pull request merged**